src/Stryker.Core/Stryker.Core/Compiling/CompilingProcess.cs

```diff
@@ -50,25 +51,40 @@ public CompilingProcessResult Compile(IEnumerable<SyntaxTree> syntaxTrees, Memor
         {
             var analyzerResult = _input.ProjectInfo.ProjectUnderTestAnalyzerResult;
 
-            var compiler = CSharpCompilation.Create(analyzerResult.Properties.GetValueOrDefault("TargetName"),
+            // add assembly info
+            StringBuilder assInfo = new StringBuilder();
+
+            assInfo.AppendLine("using System.Reflection;");
+            assInfo.AppendLine($"[assembly: AssemblyTitle(\"{analyzerResult.Properties.GetValueOrDefault("TargetName")}\")]");
+            assInfo.AppendLine("[assembly: AssemblyVersion(\"0.0.0\")]");
+            assInfo.AppendLine("[assembly: AssemblyFileVersion(\"0.0.0\")]");
+
+            syntaxTrees.ToList().Add(CSharpSyntaxTree.ParseText(assInfo.ToString(), encoding: Encoding.Default));
+
+            var compilation = CSharpCompilation.Create(analyzerResult.Properties.GetValueOrDefault("TargetName"),
                 syntaxTrees: syntaxTrees,
                 options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary,
                                                       allowUnsafe: true,
                                                       cryptoKeyFile: analyzerResult.SignAssembly ? analyzerResult.AssemblyOriginatorKeyFile : null,
                                                       strongNameProvider: analyzerResult.SignAssembly ? new DesktopStrongNameProvider() : null),
-
                 references: _input.AssemblyReferences);
+
             RollbackProcessResult rollbackProcessResult = null;
 
             // first try compiling
-            var emitResult = compiler.Emit(ms, manifestResources: analyzerResult.Resources);
+            var emitResult = compilation.Emit(ms, 
```

Comment:
Maybe we could try to call the `Emit` method only once in this file. I don't like that we have two spots where we call the actual compiler. If we merge this we are sure the emit result is always correct.

Reply:
Good idea, we keep having to call emit in multiple places every time we change it